activities.el

```diff
@@ -901,6 +914,24 @@ ignored."
            when (buffer-local-boundp variable (current-buffer))
            collect (cons variable (buffer-local-value variable (current-buffer)))))
 
+(defun activities--xref-history (&optional new-value)
+  "Return the xref history of the current activity.
+
+If not in activity, call the original xref function.
+
+Override existing value with NEW-VALUE if NEW-VALUE is set."
+  (let ((current-activity (activities-current))
+        (activity-name (when (activities-current)
+                         (activities-activity-name (activities-current)))))
+    (if current-activity
+        (progn
+          (unless (map-contains-key activities-xref-activity-history activity-name)
+            (map-put activities-xref-activity-history activity-name (xref--make-xref-history)))
```

Comment:
Also, please note the docstring of `map-elt`: the best way now is to use `(setf (map-elt ...) VALUE)`.